src/ansys/dpf/post/mesh.py

```diff
@@ -21,21 +34,419 @@ def __str__(self):
         return str(self._meshed_region).replace("Meshed Region", "Mesh")
 
     @property
-    def available_named_selections(self) -> List[str]:
-        """Returns the available named selections of the mesh."""
-        return self._meshed_region.available_named_selections
+    def named_selections(self) -> NamedSelections:
+        """Returns a dictionary of available named selections for this mesh.
+
+        Examples
+        --------
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> simulation = post.load_simulation(examples.static_rst)
+        >>> print(simulation.mesh.named_selections) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        NamedSelections dictionary with 1 named selections:
+            - '_FIXEDSU'
+        """
+        return NamedSelections(self)
 
     @property
     def node_ids(self) -> List[int]:
-        """Returns the list of node IDs in the mesh."""
+        """Returns the list of node IDs in the mesh.
+
+        Examples
+        --------
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> simulation = post.load_simulation(examples.static_rst)
+        >>> print(simulation.mesh.node_ids) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        [ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
+         25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48
+         49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72
+         73 74 75 76 77 78 79 80 81]
+        """
         return self._meshed_region.nodes.scoping.ids
 
+    @property
+    def num_nodes(self) -> int:
+        """Returns the number of nodes in the mesh.
+
+        Examples
+        --------
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> simulation = post.load_simulation(examples.static_rst)
+        >>> print(simulation.mesh.num_nodes) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        81
+        """
+        return len(self.node_ids)
```

Comment:
can we use the core API for the number of nodes here please? We will eventually have a better implementation in core and I'ld like to inherit from it here

Reply:
(same for elements)